sapi/fuzzing/fuzzer-mbstring.c

```diff
@@ -0,0 +1,85 @@
+/*
+   +----------------------------------------------------------------------+
+   | PHP Version 7                                                        |
+   +----------------------------------------------------------------------+
+   | Copyright (c) The PHP Group                                          |
+   +----------------------------------------------------------------------+
+   | This source file is subject to version 3.01 of the PHP license,      |
+   | that is bundled with this package in the file LICENSE, and is        |
+   | available through the world-wide-web at the following url:           |
+   | http://www.php.net/license/3_01.txt                                  |
+   | If you did not receive a copy of the PHP license and are unable to   |
+   | obtain it through the world-wide-web, please send a note to          |
+   | license@php.net so we can mail you a copy immediately.               |
+   +----------------------------------------------------------------------+
+   | Authors: Stanislav Malyshev <stas@php.net>                           |
+   +----------------------------------------------------------------------+
+ */
+
+
+#include "fuzzer.h"
+
+#include "Zend/zend.h"
+#include "main/php_config.h"
+#include "main/php_main.h"
+
+#include <stdio.h>
+#include <stdint.h>
+#include <stdlib.h>
+
+#include "fuzzer-sapi.h"
+
+int LLVMFuzzerTestOneInput(const uint8_t *Data, size_t Size) {
+#ifdef HAVE_MBREGEX
+	char *args[2];
+	char *data = malloc(Size+1);
+	memcpy(data, Data, Size);
+	data[Size] = '\0';
+
+	if (php_request_startup()==FAILURE) {
+		php_module_shutdown();
+		return 0;
+	}
+
+	args[0] = data;
+	args[1] = "test123";
```

Comment:
It might be fruitful to split `Data` in tow, so both parts are fuzzed.

Reply:
Hmm yes probably, not sure how fuzzer would behave in such situation... Is it a common way to deal with such things?

Reply:
The closest documentation I could find is https://github.com/google/fuzzer-test-suite/blob/master/tutorial/structure-aware-fuzzing.md#fuzzing-stateful-apis, which deals with more complicated cases where the input is broken up further. So I think generally something along these lines would be the right approach.